### ISSUE TITLE
Provide multi-arch images for other CPU arch like arm64/aarch64

### ISSUE THREAD

**Issue by xin3liang** [openebs/mayastor#1147]
**Is your feature request related to a problem? Please describe.**
Provide multi-arch images could support multi CPU architectures like arm64, amd64, etc.

**Describe the solution you'd like**
Make CI build other CPU arch images and release multi-arch images to image repo.

**Describe alternatives you've considered**
None.

**Additional context**
Many k8s images are multi-arch. It will be good to Mayastor project if it provides multi-arch images. As some other CPU arch like arm64 are becoming main stream CPU architectures.

**Comment by xin3liang** [comment]
The CI seems not all open source yet. Not sure if there some arm64 machines in the CI? And any plan to release arm64 images?

**Comment by GlennBullingham** [comment]
Hi @xin3liang.  Thanks for your request and input.

At this time the CI pipeline doesn't feature any arm64 machines, nor do the considerably more resource hungry nightly system test pipelines.  As maintainers, we're having to prioritise our investment in CI/CD and testing capability, we don't have the option of using both architectures right now.  While x86-64/AMD64 remains the predominant architecture in the data center, it will remain our priority for Mayastor.

If anyone would like to 'donate'/make available resource for this purpose, we'd be happy to have a conversation about establishing arm-based CI/CD and testing capabilities but to be clear up front, that's a lot of resource.  The system test pipeline as of version 2.1.0 will (continuously, 24x7) employ 8-node k8s clusters on bare metal, with multiple NVMe attached SSDs and 100GbE networking.  Our ambitions for long term stability and scale tests demand significantly greater levels of resource than that.

TL;DR - Mayastor's maintainers currently have no plans to provide arm-based image builds, as we have no means by which to test them to a level we consider appropriate.

Of course, anyone is (very) welcome to fork the project and provide their own images.

**Comment by xin3liang** [comment]
> Hi @xin3liang. Thanks for your request, input and contributions.
> 
> At this time the CI pipeline doesn't feature any arm64 machines, nor do the considerably more resource hungry nightly system test pipelines. As maintainers, we're having to prioritise our investment in CI/CD and testing capability, we don't have the option of using both architectures right now. While x86-64/AMD64 remains the predominant architecture in the data center, it will remain our priority for Mayastor.
> 
> If anyone would like to 'donate'/make available resource for this purpose, we'd be happy to have a conversation about establishing arm-based CI/CD and testing capabilities but to be clear up front, that's a lot of resource. The system test pipeline as of version 2.1.0 will (continuously, 24x7) employ 8-node k8s clusters on bare metal, with multiple NVMe attached SSDs and 100GbE networking. Our ambitions for long term stability and scale tests demand significantly greater levels of resource than that.
> 
> TL;DR - Mayastor's maintainers currently have no plans to provide arm-based image builds, as we have no means by which to test them to a level we consider appropriate.
> 
> Of course, anyone is (very) welcome to fork the project and provide their own images.

Hi @GlennBullingham, we've considered and talked about the Arm-based resources donation with our partners. The result is we don't have such Arm-based resources available for donating now. We just have some Arm VMs resources  available for remotely development usage which are used in some open source projects, such as Rook, Bigdata, openHPC etc. 
Anyway, hope someone have such Arm-based resources for donation in the future. And thanks for your reply.

**Comment by dm3ch** [comment]
Am I right understood that question is not  just about infrastructure for building images but also about testing and e.t.c?

**Comment by tiagolobocastro** [comment]
Yes, we could even build aarch64 container images (in fact we now build aarch64 kubectl-mayastor plugin binary) but we'd have no way of testing them :(

**Comment by dm3ch** [comment]
Am I right understood that you just don't want to release untested images and that's why you have no arm64 versions built yet?
